pkg/controller/pxc/replication.go

```diff
@@ -86,6 +92,193 @@ func (r *ReconcilePerconaXtraDBCluster) removeOutdatedServices(cr *api.PerconaXt
 	return nil
 }
 
+func (r *ReconcilePerconaXtraDBCluster) reconcileReplication(cr *api.PerconaXtraDBCluster) error {
+	if cr.Status.PXC.Ready < 1 || len(cr.Spec.PXC.ReplicationChannels) == 0 {
+		return nil
+	}
+
+	sfs := statefulset.NewNode(cr)
+
+	list := corev1.PodList{}
+	if err := r.client.List(context.TODO(),
+		&list,
+		&client.ListOptions{
+			Namespace:     cr.Namespace,
+			LabelSelector: labels.SelectorFromSet(sfs.Labels()),
+		},
+	); err != nil {
+		return errors.Wrap(err, "get pod list")
+	}
+
+	primary, err := r.getPrimaryPod(cr)
+	if err != nil {
+		return errors.Wrap(err, "get primary pxc pod")
+	}
+
+	for _, pod := range list.Items {
+		if pod.Status.PodIP == primary || pod.Name == primary {
+			primary = fmt.Sprintf("%s.%s.%s", pod.Name, sfs.Name(), cr.Namespace)
+			break
+		}
+	}
+
+	sort.Slice(list.Items, func(i, j int) bool {
+		return list.Items[i].Name > list.Items[j].Name
+	})
+
+	var primaryPod corev1.Pod
+	for _, pod := range list.Items {
+		if strings.HasPrefix(primary, fmt.Sprintf("%s.%s.%s", pod.Name, sfs.Name(), cr.Namespace)) {
+			primaryPod = pod
+			break
+		}
+	}
+
+	user := "root"
+	port := int32(3306)
+	if cr.CompareVersionWith("1.6.0") >= 0 {
+		port = int32(33062)
+	}
+
+	primaryDB, err := queries.New(r.client, cr.Namespace, cr.Spec.SecretsName, user, primaryPod.Name+"."+cr.Name+"-pxc."+cr.Namespace, port)
+	if err != nil {
+		return errors.Wrap(err, "failed to connect to pod "+primaryPod.Name)
+	}
+
+	err = primaryDB.DisableLogBin()
+	if err != nil {
+		return errors.Wrap(err, "failed to stop binlogging")
+	}
+
+	defer primaryDB.Close()
+
+	isReplica, err := primaryDB.IsReplica()
```

Comment:
it is not clear for which channel we are checking replica status.
what if we have topology like two different clusters replicated into one cluster.
I think we need to check replica status for each channel separately `SHOW REPLICA STATUS FOR CHANNEL <channel name>`